main/EDInitMod.F90

```diff
@@ -839,151 +843,174 @@ subroutine init_cohorts( site_in, patch_in, bc_in)
+             case(phosphorus_element)
 
-                case default
-                   write(fates_log(),*) 'Unspecified PARTEH module during create_cohort'
-                   call endrun(msg=errMsg(sourcefile, __LINE__))
-                end select
+                m_struct = c_struct*prt_params%phos_stoich_p1(pft,prt_params%organ_param_id(struct_organ))
+                m_leaf   = c_leaf*prt_params%phos_stoich_p1(pft,prt_params%organ_param_id(leaf_organ))
+                m_fnrt   = c_fnrt*prt_params%phos_stoich_p1(pft,prt_params%organ_param_id(fnrt_organ))
+                m_sapw   = c_sapw*prt_params%phos_stoich_p1(pft,prt_params%organ_param_id(sapw_organ))
+                m_repro  = 0._r8
+                m_store  = StorageNutrientTarget(pft,element_id,m_leaf,m_fnrt,m_sapw,m_struct)
 
-             end do
+             end select
+
+             select case(hlm_parteh_mode)
+             case (prt_carbon_allom_hyp,prt_cnp_flex_allom_hyp )
+
+                ! Put all of the leaf mass into the first bin
+                call SetState(prt_obj,leaf_organ, element_id,m_leaf,1)
+                do iage = 2,nleafage
+                   call SetState(prt_obj,leaf_organ, element_id,0._r8,iage)
+                end do
 
-             call prt_obj%CheckInitialConditions()
-             
-             call create_cohort(site_in, patch_in, pft, temp_cohort%n, temp_cohort%hite, &
-                  temp_cohort%coage, temp_cohort%dbh, prt_obj, cstatus, rstatus,        &
-                  temp_cohort%canopy_trim, temp_cohort%c_area,1,temp_cohort%crowndamage, site_in%spread, bc_in)
+                call SetState(prt_obj,fnrt_organ, element_id, m_fnrt)
+                call SetState(prt_obj,sapw_organ, element_id, m_sapw)
+                call SetState(prt_obj,store_organ, element_id, m_store)
+                call SetState(prt_obj,struct_organ, element_id, m_struct)
+                call SetState(prt_obj,repro_organ, element_id, m_repro)
 
-             deallocate(temp_cohort, stat=istat, errmsg=smsg)
-             if (istat/=0) then
-                write(fates_log(),*) 'dealloc014: fail on deallocate(temp_cohort):'//trim(smsg)
+             case default
+                write(fates_log(),*) 'Unspecified PARTEH module during create_cohort'
                 call endrun(msg=errMsg(sourcefile, __LINE__))
-             endif
-             
+             end select
+
+          end do
+
+          call prt_obj%CheckInitialConditions()
+
+          call create_cohort(site_in, patch_in, pft, temp_cohort%n, temp_cohort%hite, &
+               temp_cohort%coage, temp_cohort%dbh, prt_obj, cstatus, rstatus,        &
+               temp_cohort%canopy_trim, temp_cohort%c_area,1,temp_cohort%crowndamage, site_in%spread, bc_in)
+
+          write(fates_log(),*) 'c_area: ', temp_cohort%c_area
+          write(fates_log(),*) 'p_area: ', patch_in%area
+
+          deallocate(temp_cohort) ! get rid of temporary cohort
```

Comment:
this is a duplicate deallocate, will be harmless, but better to remove it because the following deallocation has more logging.

Reply:
Oh good catch, thanks. Deleted the duplicate.